twilio-audio-router/src/main/java/com/twilio/audio_manager/AudioDeviceSelector.java

```diff
@@ -0,0 +1,376 @@
+package com.twilio.audio_manager;
+
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.pm.PackageManager;
+import android.media.AudioAttributes;
+import android.media.AudioDeviceInfo;
+import android.media.AudioFocusRequest;
+import android.media.AudioManager;
+import android.os.Build;
+import android.os.Handler;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class AudioDeviceSelector {
```

Comment:
Add Javadoc to class declaration with code snippets that demonstrate how to use.